### ISSUE TITLE
Rename TTMetal Dialect ops to match `host_api.hpp`

### ISSUE THREAD

**Issue by nsmithtt** [tenstorrent/tt-mlir#364]
We should match the names in `host_api.hpp`.

- `DispatchOp` -> `EnqueueProgramOp`
- `HostWriteOp` -> `EnqueueWriteBufferOp`
- `HostReadOp` -> `EnqueueReadBufferOp`
- `AllocOp` -> `CreateBufferOp`
- `DeallocOp` -> `DeallocateBufferOp`

**Comment by aditya-167** [comment]
Hi I am new to TT-MLIR I would like to take this as my first PR

**Comment by nsmithtt** [comment]
Hey Aditya, that would be great! Let us know if you need any pointers.

**Comment by aditya-167** [comment]
@nsmithtt I dont see DeallocOp here, https://github.com/tenstorrent/tt-mlir/blob/8c6494cb1f4fed060073f735b1f88c5da4d187f6/lib/Dialect/TTMetal/IR/TTMetalOps.cpp  let me know if I am looking at the right thing, also just to make sure, " host_api.hpp" : https://github.com/tenstorrent/tt-metal/blob/f85ceb7c6f8392626f235a4d9b6094972cf4b55c/tt_metal/host_api.hpp this one right?

**Comment by nsmithtt** [comment]
Yeah DeallocateOp doesn't actually exist, that one is a no-op but is useful to have in the trace for debug.
